### PR TITLE
Draft for heartbeat proposal

### PR DESCRIPTION
**Pull request by yenda**
This is a proposal for a quick evolution of the protocol with 100% eventual reliability and independance from mailservers.

 It includes features that we have been discussing for a while as well as separating channel negociation from contact-request to facilitate future improvement of the protocol suggested in https://docs.google.com/document/d/1e4-lRuA8n6_btE1W8Y33RreiWYDE_L81g3OV3_6_4Bo/edit

### CODE REVIEW

File: ideas/xxx-heartbeat.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+### Iteration 1
+
+Goal Date: 2018-05-25
+
+Description: Bugs in MVP are fixed and work is merged into develop. Timeout can be configured to make testing easier
+
+## Success Metrics
+
+- 100% of messages in 1-1 chat are eventually received
```
[the remark this answers is not in the document]
Comment (yenda) [reply ideas/xxx-heartbeat.md]: We could use 99% after the two peers have been simultaneously connected for a duration of 6 timeouts with mailserver disabled. Every heartbeat with both peers online the probability of a message being lossed is lowered. After 6 timeouts the probabiliy is quasi null but not exactly null

File: ideas/xxx-heartbeat.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+## Summary
+
+Whisper being a decentralized end to end communication protocol, we don't have any server to acknowledge delivery of messages to the network. Delivery can only be confirmed by the recipient of a message, and he needs to be online to receive messages. Since 1 to 1 chats are using a secured channel negociated during contact request, we need a mechanism to detect lost messages as well as a mean to re-establish communications between peers when one recovers an account and loses access to the secured channel.
+
+The goal of this swarm is to implement a hearbeat mechanism that will fulfill this purpose, in order to improve reliability, recovery, and be independent from the mailservers for message re-delivery.
+
+## Swarm Participants
+
+> Here all roles in swarm are defined and filled, one of the contributors should
```
[the remark this answers is not in the document]
Comment (yenda) [reply ideas/xxx-heartbeat.md]: yeah it is I will remove later

File: ideas/xxx-heartbeat.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
+---
+id: 000-heartbeat
+title: Heartbeat
+status: draft
+created: 2018-05-09
+category: chat
+lead-contributor:
```
[the remark this answers is not in the document]
Comment (yenda) [reply ideas/xxx-heartbeat.md]: Yes unless someone else want to

**Comment by oskarth** [comment]
Doesn't seem unreasonable at first glance. Is there a reason this can't be done in 99-confidence swarm as a few tasks/GHIs?

**Comment by oskarth** [comment]
Closing PR as it has been open for 3 weeks. Feel free to re-open if you feel this is in error / things have changed. Will probably come up again in post-beta discussions

**Pull request closed without merging**